src/pages/settings/Payments/TransferBalancePage.js

```diff
@@ -166,70 +167,85 @@ class TransferBalancePage extends React.Component {
         const isButtonDisabled = !isTransferable || !selectedAccount;
         const errorMessage = !_.isEmpty(this.props.walletTransfer.errors) ? _.chain(this.props.walletTransfer.errors).values().first().value() : '';
 
+        const shouldShowTransferView =
+            !_.isEmpty(this.props.userWallet) &&
+            PaymentUtils.hasExpensifyPaymentMethod(this.props.cardList, this.props.bankAccountList) &&
+            this.props.userWallet.tierName &&
```

Comment:
```suggestion
```

Reply:
Makes sense. No need to validate this.props.userWallet.tierName condition as the execution only reaches here when userWallet obj is available.